internal/install/discoverer.go

```diff
@@ -0,0 +1,26 @@
+package install
```

Comment:
Refactored this file around the idea of a `discoverer` interface and related concepts.